themes/classic/_dev/js/cart.js

```diff
@@ -168,6 +168,47 @@ $(document).ready(() => {
     });
   };
 
+  //Hide 'have a code promo' text if displayed when 'close' is
+  let promoButtonsNb = $('.promo-code-button').length,
+      nameLength = 0;
+
+  function initPromoCodeButtons() {
+    $('.cancel-promo').on('click', (e) => {
+      $(e.currentTarget).hide();
+      $('.promo-code-button:not(.cancel-promo)').show();
+    });
+
+    $('.promo-code-button:not(.cancel-promo)').on('click', (e) => {
+        $(e.currentTarget).hide();
+        $('.cancel-promo').css('display','block');
+    });
+  }
+
+  function forceCancelButtonOnPromo() {
+    if ($('.block-promo-container').length != 0 && promoButtonsNb === 2) {
+      $('.cancel-promo').css('display','block');
+      $('.promo-code-button:not(.cancel-promo)').hide();
+    }
+  }
+
+  setTimeout( function () {
```

Comment:
I still don't understand why you need this `setTimeout` you are already in a `$(document).ready` code block so the DOM should be loaded already.